### ISSUE TITLE
Dashboard graph has broken y-axis range

### ISSUE THREAD

**Issue by tomasinchoo** [magento/magento2#7927]
Y-axis on dashboard graph always shows range 0 to 100 in steps of 10. That means quantity and amounts do not match actual value in graph.

### Preconditions
1. Any working setup for Magento 2 
2. Tested on 2.1.1, but it probably doesn't work in any other

### Steps to reproduce
1. Make few orders and look at the graph in admin dashboard

### Expected result
1. Values shown should be matched with y-axis value 

### Actual result
1. It always show range 10 to 100 in steps of 10 and values (qty and amounts) doesn't make sense with that values

Fortunately, that issue is easy to fix:
This code (exists in M1, not in M2) needs to be inserted on line 424 in vendor/magento/module-backend/Block/Dashboard/Graph.php (thanks to @ivanweiler)
```
                } else if ($idx == 'y') {
                    $valueBuffer[] = $indexid . ":|" . implode('|', $yLabels);   
```

**Comment by bardkalbakk** [comment]
What's the status on this bug? Still an issue in 2.1.5 believe.

**Comment by sennin32** [comment]
2.1.6 still has the same issue.

**Comment by veloraven** [comment]
Internal ticket was created - MAGETWO-69966

**Comment by Silarn** [comment]
Here's a plugin solution for this issue until the official fix:

```php
    public function __construct(
        \Magento\Backend\Helper\Dashboard\Data $dashboardData,
        \Magento\Framework\UrlInterface $url
    )
    {
        $this->_dashboardData = $dashboardData;
        $this->_urlBuilder = $url;
    }
    public function aroundGetChartUrl(\Magento\Backend\Block\Dashboard\Graph $graph, \Closure $proceed, $directUrl = true)
    {
        $query = $proceed();
        $series = $graph->getAllSeries();
        $localmaxvalue = [0];
        $localminvalue = [0];
        foreach ($series as $index => $serie) {
            $localmaxvalue[$index] = max($serie);
            $localminvalue[$index] = min($serie);
        }
        $maxvalue = max($localmaxvalue);
        $minvalue = min($localminvalue);
        $miny = 0;
        $maxy = 0;
        if ($minvalue >= 0 && $maxvalue >= 0) {
            if ($maxvalue > 10) {
                $p = pow(10, $this->_getPow($maxvalue));
                $maxy = ceil($maxvalue / $p) * $p;
            } else {
                $maxy = ceil($maxvalue + 1);
            }
        }
        $params = array('chxr' => "1,$miny,$maxy");
        // return the encoded data
        if ($directUrl) {
            return $query.'&chxr='.$params['chxr'];
        } else {
            $parts = parse_url($query);
            parse_str($parts['query'], $parsedParams);
            $params = array_merge($params, $parsedParams);
            $gaData = urlencode(base64_encode(json_encode($params)));
            $gaHash = $this->_dashboardData->getChartDataHash($gaData);
            $params = ['ga' => $gaData, 'h' => $gaHash];
            return $this->_urlBuilder->getUrl('adminhtml/*/tunnel', ['_query' => $params]);
        }
    }
    protected function _getPow($number)
    {
        $pow = 0;
        while ($number >= 10) {
            $number = $number / 10;
            $pow++;
        }
        return $pow;
    }
```

(Also thanks in part to @thaisatservermonkey)

**Comment by magento-engcom-team** [comment]
@tomasinchoo, thank you for your report.
We've created internal ticket(s) MAGETWO-69966 to track progress on the issue.

**Comment by osrecio** [comment]
Working on it, to implement solution of Silarn or tomasinchoo

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82762

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82761

**Comment by okorshenko** [comment]
The issue has been fixed in 2.2-develop branch and will be available with 2.2.2 release soon

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-82760
